addons/web_editor/static/src/js/editor/image_processing.js

```diff
@@ -334,7 +334,14 @@ async function applyModifications(img, dataOptions = {}) {
     ctx.fillRect(0, 0, result.width, result.height);
 
     // Quality
-    return result.toDataURL(mimetype, quality / 100);
+    const dataURL = result.toDataURL(mimetype, quality / 100);
+    // Every 4 bytes of base64 represent 3 bytes.
+    const newSize = dataURL.split(',')[1].length / 4 * 3;
+    const originalSize = _getImageSizeFromCache(originalSrc);
+    const isChanged = !!perspective || !!glFilter ||
+        original.width !== result.width || original.height !== result.height ||
+        original.width !== croppedImg.width || original.height !== croppedImg.height;
+    return (isChanged || originalSize >= newSize) ? dataURL : await _loadImageDataURL(originalSrc);
```

Comment:
So if there is no change and the newSize is bigger than the originalSize, return `_loadImageDataURL(originalSrc)`
But how can the newSize be > if there is your python code that already checks that ? I'm probably missing something but there is a similar condition in the image.py `if len(output_bytes) >= len(self.source) ...`

Reply:
This is to perform the same check for images that are transformed on the JS side - which is the usual case.
If I remember correctly, the python one is mostly used by the image fields, e.g. to obtain the resized images.